kura/org.eclipse.kura.protocol.can.test/src/main/java/org/eclipse/kura/protocol/can/test/CanSocketTest.java

```diff
@@ -152,13 +136,11 @@ public boolean doCanTest() {
             } catch (InterruptedException e) {
             }
         } else {
-            s_logger.info("Wait for a request");
+            logger.info("Wait for a request");
             try {
-
-                cm = this.m_canConnection.receiveCanMessage(-1, 0x7FF);
-
+                cm = this.canConnection.receiveCanMessage(-1, 0x7FF);
```

Comment:
I tend to agree. You might be lucky and dependency would be injected, or you might not and test would finish before it began.